internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/server/VeniceServerTest.java

```diff
@@ -182,6 +183,37 @@ public void testCheckBeforeJointClusterBeforeHelixInitializingCluster() throws E
     }
   }
 
+  @Test
+  public void testStartServerAndShutdownWithPartitionAssignmentVerification() {
+    try (VeniceClusterWrapper cluster = ServiceFactory.getVeniceCluster(1, 0, 0)) {
+      Properties featureProperties = new Properties();
+      featureProperties.setProperty(SERVER_ENABLE_SERVER_ALLOW_LIST, Boolean.toString(true));
+      featureProperties.setProperty(SERVER_IS_AUTO_JOIN, Boolean.toString(true));
+      cluster.addVeniceServer(featureProperties, new Properties());
+      VeniceServerWrapper server = cluster.getVeniceServers().get(0);
+      Assert.assertTrue(server.getVeniceServer().isStarted());
+      StorageService storageService = server.getVeniceServer().getStorageService();
+      StorageEngineRepository repository = storageService.getStorageEngineRepository();
+      Assert
+          .assertTrue(repository.getAllLocalStorageEngines().isEmpty(), "New node should not have any storage engine.");
+
+      // Create a storage engine.
+      String storeName = Version.composeKafkaTopic(cluster.createStore(1), 1);
+      Assert.assertEquals(repository.getAllLocalStorageEngines().size(), 1);
+      Assert.assertTrue(server.getVeniceServer().getHelixParticipationService().isRunning());
+      Assert.assertEquals(storageService.getStorageEngine(storeName).getPartitionIds().size(), 3);
+
+      server.getVeniceServer().shutdown();
+      cluster.getControllerClient().deleteStore(storeName);
+
+      cluster.stopVeniceServer(server.getPort());
```

Comment:
Is this needed since you're already doing the shutdown on line 206?

Reply:
Yes, the storage service cannot be obtained if I don't use `cluster.stopVeniceServer` and `cluster.restartVeniceServer.`

Reply:
I'm not sure I understand. 

Shutdown is performed on line 206, and on line 209 stop is called on the server.